core/app/models/spree/order.rb

```diff
@@ -787,7 +787,7 @@ def after_cancel
     end
 
     def send_cancel_email
-      OrderMailer.cancel_email(self.id).deliver_later
+      OrderMailer.cancel_email(self).deliver_later
```

Comment:
it looks like this is something they have started supporting: https://github.com/rails/rails/tree/master/activejob#globalid-support